docs/v2_design.md

```diff
@@ -0,0 +1,179 @@
+```
+
+### Derived (unnamed) quantity/dimension and derived (unnamed) unit
+
+```cpp
+auto q = 10 * si::length[m] / (2 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(q),
+              quantity<reference<derived_dimension<isq::length_dim, per<isq::time_dim>>,
+                       derived_unit<si::metre, per<si::second>>>{}, int>>);
+```
+
+### Base quantity as a result of dimensional transformation
+
+```cpp
+auto d = 5 * si::speed[m / s] * (5 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(d), quantity<reference<isq::length_dim, si::metre>{}, int>>);
+```
+
+### Dimensionless
+
+```cpp
+auto q = 20 * si::speed[m / s] / (10 * si::length[m]) * (5 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(q), quantity<reference<one_dim, one>{}, int>>);
+```
+
+---
+
+### Comparison
+
+- Should the below compile?
+- Should it be true?
+
+```cpp
+// Named and derived dimensions (same units)
+bool b1 = 10 * si::length[m] / (2 * si::time[s]) == 5 * si::speed[m / s];
+
+// Named and derived dimensions (different units)
+bool b2 = 10 / (2 * si::time[s]) == 5 * si::frequency[Hz];
+
+// Different named dimensions
+auto b3 = 5 * si::activity[Bq] == 5 * si::frequency[Hz];
+```
+
+---
+
+## Arithmetics
+
+- Should the below compile?
+- If yes, what should be the final type?
+
+```cpp
+// Named and derived dimensions (same units)
+auto q1 = 10 * si::length[m] / (2 * si::time[s]) + 5 * si::speed[m / s];
+
+// Named and derived dimensions (different units)
+auto q2 = 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+
+// Different named dimensions
+auto q3 = 5 * si::activity[Bq] + 5 * si::frequency[Hz];
+
+// ???
+auto q3a = 5 * si::activity[Bq] + 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+```
+
+---
+
+## Comparisons: Speed
+
+```cpp
+// same dimension type & different unit
+static_assert(1000 * si::length[m] == 1 * si::length[km]);
+// different dimension type (derived unnamed vs named)
+static_assert(20 * si::length[m] / (2 * si::time[s]) == 10 * si::speed[m]); // ???
+
+static_assert(isq::length_dim == isq::length_dim);
+// derived unnamed vs named
+static_assert(equivalent(isq::length_dim / isq::time_dim, isq::speed_dim));
+static_assert(isq::length_dim / isq::time_dim != isq::speed_dim);           // ???
+// both sides return the same base dimension (at least for now)
+static_assert(equivalent(isq::speed_dim * isq::time_dim, isq::length_dim));
+static_assert(isq::speed_dim * isq::time_dim == isq::length_dim);           // ???
+// NOTE: "magnetomotive force" derived quantity is measured in ampere (which is a base unit)
+// Should the result of the above be just `isq::length_dim` or
+// `derived_dimension<isq::length_dim>` that can me converted to isq::length_dim?
+// In the latter case we would be able to make the above `false` and be able to
+// make "magnetomotive force" a dimension distinguishable from "electric current"
+
+static_assert(equivalent(si::metre, si::kilometre));
```

Comment:
If this assert should pass, `equivalent` is a bad name. They may have the same base unit, but they're not what falls under "equivalent" as the data structures (1, metre) and (10³, metre). Maybe if they were projected on the base unit, in the `std::ranges` sense.

Reply:
Yes, I just took a shower just before I read your comment and thought about exactly the same. I think that I should just use the `convertible` term as this will be less surprising and more standard-compliant.

Reply:
Is there a place for `equivalent` in the library or should just all the instances should be renamed to `convertible`? What about explicit and implicit conversions? Should we introduce `implicitly_convertible` and `explicitly_convertible` as well? `km -> m` is an implicit conversion but `m -> km` should probably be considered an explicit conversion as it is truncating operation.